src/info/Info.ts

```diff
@@ -0,0 +1,135 @@
+            const rv = userAgent.match(/rv:(\d+)/) || [];
+            version = rv[1];
+        }
+        if (match[1] === 'Chrome') {
+            const temp = userAgent.match(/\b(OPR|Edge)\/(\d+)/);
+            if (temp != null) version = temp[2];
+        }
+        return version || "Unknown";
+    }
+
+    public static GetSystem(): string {
+        const platform = navigator.platform;
+        if (/Android/.test(navigator.userAgent)) {
+            return "Android";
+        } else if (/iPhone|iPad|iPod/.test(navigator.userAgent)) {
+            return "iOS";
+        } else if (platform.startsWith("Win")) {
+            return "Windows";
+        } else if (platform.startsWith("Mac")) {
+            return "MacOS";
+        } else if (platform.startsWith("Linux")) {
+            return "Linux";
+        } else {
+            return "Unknown";
+        }
+    }
+
+    public static GetSystemVersion(): string {
+        const userAgent = navigator.userAgent;
+        const match = userAgent.match(/(Windows NT|Mac OS X|Android|CPU iPhone OS|CPU OS|Linux) ([\d._]+)/);
+        if (match) {
+            return match[2].replace(/_/g, '.');
+        }
+        return "Unknown";
+    }
+
+    public static async GetSupportsWebXR(): Promise<boolean> {
+        return new Promise((resolve) => {
+            if (this._supportsWebXR !== null) {
+                resolve(this._supportsWebXR);
+                return;
+            }
+
+            if (!navigator.xr) {
+                resolve(false);
+                return;
+            }
+            // Check if immersive-vr session mode is supported
+            navigator.xr.isSessionSupported('immersive-ar')
+                .then((supported) => {
+                    if (supported) {
+                        this._supportsWebXR = true;
+                        resolve(true);
+                    } else {
+                        resolve(false);
+                    }
+                }).catch(() => {
+                    resolve(false);
+                });
+        });
+
+    }
+
+    public static GetSupportsARQuickLook(): boolean {
+        const userAgent = navigator.userAgent;
+
+        // Check if the device is running iOS
+        const isIOS = /iPad|iPhone|iPod/.test(userAgent) && !(window as unknown as Window & { MSStream?: string }).MSStream;
+        if (!isIOS) {
+            return false;
+        }
+
+        // Extract iOS version
+        const match = userAgent.match(/OS (\d+)_/);
+        if (!match || match.length < 2) {
+            return false;
+        }
+        const iOSVersion = parseInt(match[1], 10);
+
+        // Minimum iOS version for QuickLook support
+        const minQuickLookVersion = 12;
+
+        // Check if the iOS version is supported
+        if (iOSVersion < minQuickLookVersion) {
+            return false;
+        }
+
+        // Check for specific browser
+        const isSafari = /^((?!chrome|android).)*safari/i.test(userAgent);
```

Comment:
I guess we can have only one const isSupportedBrowser, then check with a || b || c  || d

Reply:
Yes, agree, will be changed